internal/server/server.go

```diff
@@ -24,15 +27,38 @@ type Template interface {
 
 func New(logger *log.Logger, client Client, templates map[string]*template.Template, prefix, siriusURL, siriusPublicURL string, webDir string) http.Handler {
 	wrap := errorHandler(logger, templates["error.gotmpl"], prefix, siriusPublicURL)
+	systemAdminOnly := systemAdminOnly(client)
 
 	mux := http.NewServeMux()
 	mux.Handle("/", http.RedirectHandler(prefix+"/my-details", http.StatusFound))
 	mux.Handle("/health-check", healthCheck())
-	mux.Handle("/users", wrap("listUsers", listUsers(logger, client, templates["users.gotmpl"], siriusURL)))
-	mux.Handle("/my-details", wrap("myDetails", myDetails(logger, client, templates["my-details.gotmpl"], siriusURL)))
-	mux.Handle("/my-details/edit", wrap("editMyDetails", editMyDetails(logger, client, templates["edit-my-details.gotmpl"], siriusURL)))
-	mux.Handle("/change-password", wrap("changePassword", changePassword(logger, client, templates["change-password.gotmpl"], siriusURL)))
-	mux.Handle("/add-user", wrap("addUser", addUser(logger, client, templates["add-user.gotmpl"], siriusURL)))
+
+	mux.Handle("/users",
+		wrap("listUsers",
+			systemAdminOnly(
+				listUsers(client, templates["users.gotmpl"], siriusURL))))
+
+	mux.Handle("/my-details",
+		wrap("myDetails",
+			myDetails(client, templates["my-details.gotmpl"], siriusURL)))
+
+	mux.Handle("/my-details/edit",
+		wrap("editMyDetails",
+			editMyDetails(client, templates["edit-my-details.gotmpl"], siriusURL)))
+
+	mux.Handle("/change-password",
+		wrap("changePassword",
+			changePassword(client, templates["change-password.gotmpl"], siriusURL)))
+
+	mux.Handle("/add-user",
```

Comment:
Not a blocker for this PR, but I'd prefer the URL structure of `/user/add`, `user/edit/123`. Admittedly that's not what's on the prototype, but URLs are a bit more fiddly there.